### PR TITLE
Introduce gosec for security checks

### PR DESCRIPTION
**Pull request by JAORMX**
gosec does static code analysis and checks for common security issues in
golang codebases.

This PR introduces a script that will run gosec similarly to other check
tools.

### CODE REVIEW

File: hack/gosec.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+# Example:  ./hack/gosec.sh
+set -x
+
+if [ "$IS_CONTAINER" != "" ]; then
+  go get github.com/securego/gosec/cmd/gosec
```
Comment (wking) [inline hack/gosec.sh]: Our [only precedent for a local install][1] in a `hack/` script is gated on command existence and uses `go get -u`, both of which seem like good ideas to me.

[1]: https://github.com/openshift/installer/blob/d89a59f68e2d7786f585242925f040fc645c9a54/hack/verify-vendor.sh#L4

File: hack/gosec.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  gosec -severity medium -confidence medium -exclude G304 ./cmd/... ./data/... ./pkg/... "${@}"
+else
+  podman run --rm \
+    --env IS_CONTAINER=TRUE \
+    --volume "${PWD}:/go/src/github.com/openshift/installer:z" \
+    --workdir /go/src/github.com/openshift/installer \
+    docker.io/openshift/origin-release:golang-1.12 \
```
Comment (wking) [inline hack/gosec.sh]: All of our other `hack/` scripts use `golang-1.10`.  I'd rather stick with that if possible, and then bump them all when the repo as a whole moves to 1.12 (or any other Go version).

File: hack/gosec.sh (hunk 1 above)
Comment (JAORMX) [reply hack/gosec.sh]: ack, will change this.

File: hack/gosec.sh (hunk 2 above)
Comment (JAORMX) [reply hack/gosec.sh]: It doesn't work with that image... I only got it to work with 1.12

File: hack/gosec.sh (hunk 1 above)
Comment (JAORMX) [reply hack/gosec.sh]: done.

File: hack/gosec.sh (hunk 2 above)
Comment (JAORMX) [reply hack/gosec.sh]: Is using 1.10 still required? I noticed other openshift repos using 1.12

**Comment by JAORMX** [comment]
any more feedback for this?

**Comment by abhinavdahiya** [comment]
/close

closing without prejudice due to age. Please consider opening an issue or [enhancement](https://github.com/openshift/enhancements/) to discuss and bring consensus on the fix, or if you think this is still important in current state feel free to reopen.

**Comment by JAORMX** [comment]
@abhinavdahiya the commit was still active though and even got a review 3 days ago.

**Pull request closed without merging**